packages/jaeger-ui/src/components/TracePage/index.tsx

```diff
@@ -330,10 +332,15 @@ export class TracePageImpl extends React.PureComponent<TProps, TState> {
         findCount = spanFindMatches ? spanFindMatches.size : 0;
       }
     }
-    //  =  ?  : null;
-    // const spanFindMatches = !traceGraphView ? this._filterSpans(uiFind || '', _get(trace, 'data.spans')) : null;
+
     const isEmbedded = Boolean(embedded);
     const headerProps = {
+      focusUiFind: () => {
+        if (trace.data) {
+          focusUiFind(trace.data, uiFind);
+          setTimeout(this._scrollManager.scrollToFirstVisibleSpan);
```

Comment:
The scroll destination changes as a result of dispatching `focusUiFind`. This `setTimeout` achieves the desired result but definitely seems like the wrong way to achieve this.
@tiffon how would you recommend scrolling to the first match only after the rows have been expanded/collapsed and had their children shown/hidden?

Reply:
Unfortunately an instance property wouldn't work because the component that updates as a result of `focusUiFind` is not `TracePage/index.tsx`. I got around this by adding a property to the `detailStates` Map and `childrenHiddenIDs` Set made by `calculateHiddenIdsAndDetailStates` in `TracePage/duck.tsx` and reading that property in `componentDidUpdate` in `TraceTimelineViewer/VirtualizedTraceView.tsx`
It's slightly unorthodox in principle, but not so bad in practice.
[Here is the commit that solely addresses setTimeout](https://github.com/jaegertracing/jaeger-ui/pull/367/commits/ebf9473b9613815e861723b77e69ad7eb7c577e2)
TODO: more tests and icons